wp-includes/class-wp-https-ui.php

```diff
@@ -0,0 +1,336 @@
+			sprintf(
+				'<a href="%s">%s</a>',
+				__( 'https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content/How_to_fix_website_with_mixed_content', 'pwa' ),
+				__( 'More details', 'pwa' )
+			)
+		);
+
+		/**
+		 * Add class="hidden" to this <div> if the 'HTTPS Upgrade' checkbox isn't checked.
+		 * This insecure content UI does not apply if the user isn't upgrading to HTTPS,
+		 * as there will be no need to upgrade insecure requests.
+		 */
+		?>
+		<div id="<?php echo esc_attr( $insecure_content_id ); ?>" <?php echo ! $upgrade_https_value ? 'class="hidden"' : ''; ?>>
+			<p class="description">
+				<?php echo wp_kses_post( $description ); ?>
+			</p>
+			<ul class="<?php echo esc_attr( $insecure_urls_class ); ?>">
+				<?php
+				for ( $i = 0; $i < self::NUMBER_INITIAL_URLS; $i++ ) :
+					if ( ! isset( $all_insecure_urls[ $i ] ) ) :
+						break;
+					endif;
+
+					$url           = $all_insecure_urls[ $i ];
+					$truncated_url = $this->get_truncated_url( $url );
+					?>
+					<li><a href="<?php echo esc_attr( $url ); ?>"><?php echo esc_html( $truncated_url ); ?></a></li>
+				<?php endfor; ?>
+			</ul>
+			<?php
+			// Output the <ul> elements that display on clicking 'Show more'.
+			while ( isset( $all_insecure_urls[ $i ] ) ) :
+				?>
+				<ul class="<?php echo esc_attr( $insecure_urls_class ); ?> hidden">
+					<?php
+					for ( $j = 0; $j < self::NUMBER_URLS_IN_EACH_UL; $j++ ) :
+						if ( ! isset( $all_insecure_urls[ $i ] ) ) {
+							break;
+						}
+						$url           = $all_insecure_urls[ $i++ ];
+						$truncated_url = $this->get_truncated_url( $url );
+						?>
+						<li><a href="<?php echo esc_attr( $url ); ?>"><?php echo esc_html( $truncated_url ); ?></a></li>
+					<?php endfor; ?>
+				</ul>
+			<?php endwhile; ?>
+			<?php if ( $total_urls_count > self::NUMBER_INITIAL_URLS ) : ?>
+				<button id="<?php echo esc_attr( $show_more_button_id ); ?>" class="button button-secondary"><?php esc_html_e( 'Show more', 'pwa' ); ?></button>
+			<?php endif; ?>
+		</div>
+		<script>
+			( function ( $ ) {
+				// On checking 'Upgrade to secure connection,' toggle the display of the insecure URLs, as they don't apply unless it's checked.
+				$( 'input[type=checkbox][name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>"]' ).on( 'change', function() {
+					$( '#<?php echo esc_attr( $insecure_content_id ); ?>' ).toggleClass( 'hidden' );
+				} );
+
+				$( '#<?php echo esc_attr( $show_more_button_id ); ?>' ).on( 'click', function( event ) {
+					event.preventDefault();
+					$( '.<?php echo esc_attr( $insecure_urls_class ); ?>.hidden' ).first().removeClass( 'hidden' );
+
+					// If there are no more insecure URLs that are hidden, hide the 'Show more' button.
+					if ( ! $( '.<?php echo esc_attr( $insecure_urls_class ); ?>.hidden' ).length ) {
+						$( this ).addClass( 'hidden' );
+					}
+				} );
+			} )( jQuery );
+		</script>
+		<style>
+			.<?php echo esc_attr( $insecure_urls_class ); ?> {
```

Comment:
HTML entity-encoding doesn't work in CSS. Probably better to use `sanitize_key()` if really needed. But since these are the strings defined in the method itself, it's not essential to escape.

Reply:
Sure, https://github.com/xwp/pwa-wp/pull/36/commits/9e969c0475549792f504528a6e1041b9440a2262 removes this `esc_attr()`.